src/js/BookReader/ImageCache.js

```diff
@@ -32,6 +32,37 @@ export class ImageCache {
     return this._createImage(index, reduce);
   }
 
+  /**
+   * Checks if image has been loaded
+   * @param {String|Number} index - page index
+   * @returns {Boolean}
+   */
+  imageLoaded(index) {
+    return !!this.cache[index]?.loaded;
+  }
+
+  /**
+   * Checks if image's reduce
+   * @param {String|Number} index - page index
+   * @returns {Number} reduce
+   */
+  imageReduce(index) {
+    return this.cache[index]?.reduce;
+  }
+
+  /**
+   * Checks if image's cache status
+   * @param {String|Number} index - page index
+   * @returns {{ index: Number, inCache: (Object|undefined), reduce: Number, loaded: Boolean }}
```

Comment:
You can remove this; it'll be able to deduce. (Less likely to fall out of sync with the code if we don't duplicate it).
```suggestion
```